shared/teams/team/rows/empty-row.tsx

```diff
@@ -0,0 +1,133 @@
+  members: 'Add/Invite people',
+  subteams: 'Create a subteam',
+}
+
+const useSecondaryAction = (props: Props) => {
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+  const teamID = props.teamID
+  const onSecondaryAction = () => {
+    switch (props.type) {
+      case 'members':
+        // TODO: different behavior for channels - but waiting on the add members button in parent to be implemented
+        dispatch(TeamsGen.createStartAddMembersWizard({teamID}))
+        break
+      case 'subteams':
+        dispatch(TeamsGen.createLaunchNewTeamWizardOrModal({subteamOf: teamID}))
+        break
+      case 'channelsFew':
+        dispatch(nav.safeNavigateAppendPayload({path: [{props: {teamID}, selected: 'chatCreateChannel'}]}))
+        break
+      case 'channelsEmpty':
+        // TODO(Y2K-1700)
+        dispatch(nav.safeNavigateAppendPayload({path: [{props: {teamID}, selected: 'teamCreateChannels'}]}))
+        break
+    }
+  }
+  return onSecondaryAction
+}
+
+// TODO: decide what to do about the <5 channel empty state
+const EmptyRow = (props: Props) => {
+  const {conversationIDKey, teamID} = props
+  const teamMeta = Container.useSelector(state => Constants.getTeamMeta(state, teamID))
+  const notIn = teamMeta.role === 'none' || props.notChannelMember
+  const you = Container.useSelector(state => state.config.username)
+
+  const dispatch = Container.useDispatch()
+  const onSecondaryAction = useSecondaryAction(props)
+  const onAddSelf = () =>
+    dispatch(
+      conversationIDKey
+        ? Chat2Gen.createJoinConversation({conversationIDKey})
+        : TeamsGen.createAddToTeam({
+            sendChatNotification: false,
+            teamID,
+            users: [{assertion: you, role: 'admin'}],
+          })
+    )
+  const waiting = Container.useAnyWaiting(Constants.addMemberWaitingKey(teamID, you))
+
+  const teamOrChannel = props.conversationIDKey ? 'channel' : 'team'
+  const teamOrChannelName = props.conversationIDKey ? 'This channel' : teamMeta.teamname
+  return (
+    <Kb.Box2 direction="vertical" gap="small" alignItems="center" style={styles.container} fullWidth={true}>
+      <Kb.Box2 direction="horizontal">
+        <Kb.Icon type={icon[props.type]} />
+      </Kb.Box2>
+      <Kb.Text type="BodySmall" center={true} style={styles.text}>
+        {props.type === 'members'
+          ? notIn
+            ? `${teamOrChannelName} doesn't have any members`
+            : `You are the only member in this ${teamOrChannel}.`
+          : props.type === 'channelsEmpty'
+          ? `${teamMeta.teamname} is a small team.
+Make it a big team by creating chat channels.`
+          : props.type === 'channelsFew'
+          ? 'Channels can be joined by anyone, unlike subteams.'
+          : props.type === 'subteams'
+          ? 'Subteams are cryptographically distinct, and can welcome people who aren’t elsewhere in your team hierarchy.'
+          : undefined}
```

Comment:
Probably better for this chunk here to be a `switch` on `type` in a helper function.